core/src/main/scala/com/salesforce/op/stages/impl/preparators/MinVarianceFilter.scala

```diff
@@ -0,0 +1,258 @@
+    val minVar = $(minVariance)
+    // Including dummy params in reasonsToRemove to allow re-use of `ColumnStatistics`
+    for {
+      col <- stats
+      reasons = col.reasonsToRemove(
+        minVariance = minVar,
+        minCorrelation = 0.0,
+        maxCorrelation = 1.0,
+        maxCramersV = 1.0,
+        maxRuleConfidence = 1.0,
+        minRequiredRuleSupport = 1.0,
+        removeFeatureGroup = false,
+        protectTextSharedHash = true,
+        removedGroups = Seq.empty
+      )
+      if reasons.nonEmpty
+    } yield {
+      logWarning(s"Removing ${col.name} due to: ${reasons.mkString(",")}")
+      col
+    }
+  }
+
+  override def fitFn(data: Dataset[OPVector#Value]): UnaryModel[OPVector, OPVector] = {
+    // Set the desired log level
+    if (isSet(logLevel)) {
+      Option(log).collect { case l: Log4jLoggerAdapter =>
+        LogManager.getLogger(l.getName).setLevel(Level.toLevel($(logLevel)))
+      }
+    }
+    val removeBad = $(removeBadFeatures)
+
+    logInfo("Getting vector rows")
+    val vectorRows: RDD[OldVector] = data.rdd.map {
+      case sparse: SparseVector => OldVectors.sparse(sparse.size, sparse.indices, sparse.values)
+      case dense: DenseVector => OldVectors.dense(dense.toArray)
+    }.persist()
+
+    logInfo("Calculating columns stats")
+    val colStats = Statistics.colStats(vectorRows)
+    val count = colStats.count
+    require(count > 0, "Sample size cannot be zero")
+
+    val featureSize = vectorRows.first().size
+    require(featureSize > 0, "Feature vector passed in is empty, check your vectorizers")
+
+    // handle any possible serialization errors if users give us wrong metadata
+    val vectorMeta = try {
+      OpVectorMetadata(getInputSchema()(in1.name))
+    } catch {
+      case e: NoSuchElementException =>
+        throw new IllegalArgumentException("Vector input metadata is malformed: ", e)
+    }
+
+    require(featureSize == vectorMeta.size,
+      s"Number of columns in vector metadata (${vectorMeta.size}) did not match number of columns in data" +
+        s"($featureSize), check your vectorizers \n metadata=$vectorMeta")
+    val vectorMetaColumns = vectorMeta.columns
+    val featureNames = vectorMetaColumns.map(_.makeColName())
+
+    logInfo("Logging all statistics")
+    val stats = makeColumnStatistics(vectorMetaColumns, colStats)
+    stats.foreach { stat => logInfo(stat.toString) }
+
+    logInfo("Calculating features to remove")
+    val toDropFeatures = if (removeBad) getFeaturesToDrop(stats) else Array.empty[ColumnStatistics]
+    val toDropSet = toDropFeatures.flatMap(_.column).toSet
+    val outputFeatures = vectorMetaColumns.filterNot { col => toDropSet.contains(col) }
+    val indicesToKeep = outputFeatures.map(_.index)
+
+    val outputMeta = OpVectorMetadata(getOutputFeatureName, outputFeatures, vectorMeta.history)
+
+    val summaryMetadata = {
+      val featuresStatistics = new SummaryStatistics(colStats, sample = 1.0)
```

Comment:
In general we ant to have a case class for metadata with a toMetadata and fromMetadata method to make working with these easier